core/templates/domain/creator_dashboard/creator-dashboard-backend-api.service.ts

```diff
@@ -21,19 +21,181 @@ import { downgradeInjectable } from '@angular/upgrade/static';
+import { LoggerService } from 'services/contextual/logger.service';
+import { SuggestionsService } from
+  'services/suggestions.service';
+
+interface CreatorDashboardDataBackendDict {
+  'dashboard_stats': CreatorDashboardStatsBackendDict;
+  'last_week_stats': CreatorDashboardStatsBackendDict;
+  'display_preference': 'card' | 'list';
+  'subscribers_list': SubscriberSummaryBackendDict[];
+  'threads_for_created_suggestions_list': FeedbackThreadBackendDict[];
+  'threads_for_suggestions_to_review_list': FeedbackThreadBackendDict[];
+  'created_suggestions_list': SuggestionBackendDict[];
+  'suggestions_to_review_list': SuggestionBackendDict[];
+  'explorations_list': CreatorExplorationSummaryBackendDict[];
+  'collections_list': CollectionSummaryBackendDict[];
+  // Here topic summary dicts are optional because they are present in response
+  // only if new structrures are enabled.
+  'topic_summary_dicts'?: TopicSummaryBackendDict[];
+}
+
+interface CreatorDashboardData {
+  dashboardStats: CreatorDashboardStats;
+  lastWeekStats: CreatorDashboardStats;
+  displayPreference: 'card' | 'list';
+  subscribersList: ProfileSummary[];
+  threadsForCreatedSuggestionsList: FeedbackThread[];
+  threadsForSuggestionsToReviewList: FeedbackThread[];
+  createdSuggestionsList: Suggestion[];
+  suggestionsToReviewList: Suggestion[];
+  createdSuggestionThreadsList: SuggestionThread[];
+  suggestionThreadsToReviewList: SuggestionThread[];
+  explorationsList: CreatorExplorationSummary[];
+  collectionsList: CollectionSummary[];
+  topicSummaries?: TopicSummary[];
+}
+
 @Injectable({
   providedIn: 'root'
 })
 export class CreatorDashboardBackendApiService {
-  constructor(private http: HttpClient) {}
+  constructor(
+    private http: HttpClient,
+    private collectionSummaryObjectFactory:
+    CollectionSummaryObjectFactory,
+    private creatorDashboardStatsObjectFactory:
+    CreatorDashboardStatsObjectFactory,
+    private creatorExplorationSummaryObjectFactory:
+    CreatorExplorationSummaryObjectFactory,
+    private profileSummaryObjectFactory: ProfileSummaryObjectFactory,
+    private feedbackThreadObjectFactory: FeedbackThreadObjectFactory,
+    private suggestionObjectFactory: SuggestionObjectFactory,
+    private suggestionThreadObjectFactory: SuggestionThreadObjectFactory,
+    private suggestionsService: SuggestionsService,
+    private topicSummaryObjectFactory: TopicSummaryObjectFactory,
+    private loggerService: LoggerService) {}
+
+  _getSuggestionThreads(
+      feedbackDicts: FeedbackThreadBackendDict[],
+      suggestionDicts: SuggestionBackendDict[]): SuggestionThread[] {
+    var numberOfSuggestions = feedbackDicts.length;
+    var suggestionThreads: SuggestionThread[] = [];
+
+    for (var i = 0; i < numberOfSuggestions; i++) {
+      if (suggestionDicts.length !== numberOfSuggestions) {
+        this.loggerService.error(
+          'Number of suggestions does not match number of suggestion threads');
+      }
```

Comment:
Move this away from the for loop, I do not see why it needs to be inside?

Reply:
done. Actually I did it exactly the way it was done earlier.